src/cider/exec/nextgen/Nextgen.h

```diff
@@ -27,11 +27,12 @@
 
 namespace cider::exec::nextgen {
 
-using QueryFunc = void (*)(int8_t*, int8_t*);
+using QueryFunc = int32_t (*)(int8_t*, int8_t*);
```

Comment:
question: return value is error code?

Reply:
compared with using a variable to convey error code, @bigPYJ1151 thought using the return value is a better choice.